### ISSUE TITLE
"Laser" pointer?

### ISSUE THREAD

**Issue by TheCherry** [artumino/PapyrusVR#13]
Hi PapyrusVR Team,

will this Lib have support for a "laserpointer" (like the map menu) ? So that we can develop SkyUI and the Perk menu with a mouse like support? That will be awesome.

Greetings

**Comment by artumino** [comment]
So, this problem is currently on the list of things that we want to eventually have but has some issues that needs to be solved (and a good bit of testing is involved). On the top of my head:

- For 2D menus (like SkyUI and the default menus), we need to detect the intersection between menu plane and the controller forward vector. Once we can figure out distance and size of the plane from the player, possibly in SteamVR space, calculating the intersection point is trivial. This might actually be already possible IF every menu panel has a fixed size and appears at a fixed distance from the player in SteamVR space.
- For 3D menus the solution above can't really work and might actually be a lot more complicated, so I'll suggest starting from 2D menus
- Once we can figure out the 2D point on the mouse plane where the controller is pointing we still need to send that event to Scaleform as Skyrim would in the pancake version, that might actually be easy but I don't have any experience with it...
Maybe using some of the MenuEventHandler that we were discussing in issue #010 would be possible.

So, to sum it up 2D laser interactions might be possible but we need to do a good bit of testing to figure out where the panels appear relative to the player, their dimensions and their resolution. We also need to test if we can send fake mouse events to Scaleform and if SkyrimVR still handles them. If this isn't the case than the thing gets more complicated cause we either expose some events and leave the work to the mod author or we need to come up with a different solution.
3D interactions for now are out of discussion...
